plugins/bulk_upload/drop_folder_xml/plugins/providers/ap_feed/ApFeedDropFolderPlugin.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @package plugins.ApFeedDropFolder
+ */
+class ApFeedDropFolderPlugin extends KalturaPlugin implements IKalturaPlugin, IKalturaPending, IKalturaObjectLoader, IKalturaEnumerator, IKalturaApplicationTranslations
+{
+	const PLUGIN_NAME = 'ApFeedDropFolder';
+	
+	const DROP_FOLDER_PLUGIN_NAME = 'dropFolder';
+	/**
+	 * @return array
+	 */
+	public static function getTranslations($locale)
+	{
+		$array = array();
+		
+		$langFilePath = __DIR__ . "/config/lang/$locale.php";
+		if(!file_exists($langFilePath))
+		{
+			$default = 'en';
+			$langFilePath = __DIR__ . "/config/lang/$default.php";
+		}
+		
+		KalturaLog::info("Loading file [$langFilePath]");
+		$array = include($langFilePath);
+		
+		return array($locale => $array);
+	}
+	
+	/**
+	 * Returns a list of enumeration class names that implement the baseEnumName interface.
+	 * @param string $baseEnumName the base implemented enum interface, set to null to retrieve all plugin enums
+	 * @return array<string> A string listing the enum class names that extend baseEnumName
+	 */
+	public static function getEnums($baseEnumName = null)
+	{
+		if (!$baseEnumName)
+		{
+			return array('ApFeedDropFolderType');
+		}
+		if ($baseEnumName == 'DropFolderType')
```

Comment:
1. Use === for string match
2. For 'DropFolderType' use KalturaDropFolderType::getEnumClass()

Reply:
@MosheMaorKaltura the reason this is done this way (in all of our plugins - not just this one) is because the IKalturaEnumerator::getEnums method is called from several endpoints - admin console, batch and generator. Each context has a different classmap loaded and in some cases this method is called in order to build the class map. In order to reference this value as a class enum, the class defining the enum needs to be part of the classmap, leading to an exception being thrown.